extensions/amp-form/0.1/amp-form.js

```diff
@@ -176,9 +188,75 @@ export class AmpForm {
 }
 
 
+function reportValidity(state) {
+  reportFormValidity(state.form);
+}
+
+/**
+ * Reports validity for the first invalid input - if any.
+ * @param {!HTMLFormElement} form
+ * @private
+ */
+function reportFormValidity(form) {
+  const inputs = toArray(form.querySelectorAll('input,select,textarea'));
+  for (let i = 0; i < inputs.length; i++) {
+    if (!inputs[i].checkValidity()) {
+      reportInputValidity(inputs[i]);
+      break;
+    }
+  }
+}
+
+
+/**
+ * Revalidates the currently focused input after a change.
+ * @param {!KeyboardEvent} event
+ * @private
+ */
+function onInvalidInputKeyUp_(event) {
+  validationBubble.hide();
```

Comment:
This will schedule two `VSync` tasks, when only one is ever necessary. Either it's valid, and we can hide it, or it's invalid, and we need to update and center it.